server/src/main/java/org/opensearch/rest/action/admin/indices/RestViewAction.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin.indices;
+
+import org.opensearch.action.admin.indices.view.CreateViewAction;
+import org.opensearch.action.admin.indices.view.DeleteViewAction;
+import org.opensearch.action.admin.indices.view.GetViewAction;
+import org.opensearch.action.admin.indices.view.ListViewNamesAction;
+import org.opensearch.action.admin.indices.view.SearchViewAction;
+import org.opensearch.action.admin.indices.view.UpdateViewAction;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.ValidationException;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.xcontent.XContentParser;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.NamedRoute;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestCancellableNodeClient;
+import org.opensearch.rest.action.RestStatusToXContentListener;
+import org.opensearch.rest.action.RestToXContentListener;
+import org.opensearch.rest.action.search.RestSearchAction;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.function.IntConsumer;
+
+import static org.opensearch.rest.RestRequest.Method.DELETE;
+import static org.opensearch.rest.RestRequest.Method.GET;
+import static org.opensearch.rest.RestRequest.Method.POST;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+/** All rest handlers for view actions */
+@ExperimentalApi
+public class RestViewAction {
+
+    public static final String VIEW_NAME = "view_name";
+    public static final String VIEW_NAME_PARAMETER = "{" + VIEW_NAME + "}";
+
+    /** Handler for create view */
+    @ExperimentalApi
+    public static class CreateViewHandler extends BaseRestHandler {
+
+        @Override
+        public List<Route> routes() {
+            return List.of(new NamedRoute.Builder().path("/views").method(POST).uniqueName(CreateViewAction.NAME).build());
+        }
```

Comment:
I'm curious about the choice to do a dedicated create action driven by POST, rather than doing an upsert driven by PUT. (I haven't done a lot of REST APIs, but the ones I've worked on have tended to do the latter. I'm wondering if I've been doing it wrong.)

Reply:
I've always thought of creation as POST operation, and that PUT's should 404 - this prevents unintentional overwriting of existing resources.  However, I'm not married to this pattern if we've got other examples that would be better to snap too.

Reply:
Your understanding is consistent with [this guide](https://hub.packtpub.com/what-are-rest-verbs-and-status-codes-tutorial/), for example. Copy-pasting the helpful reference table from that article:

REST Verb | Action | Success | Failure
-- | -- | -- | --
GET | Fetches a record or set of resources from the server | 200 | 404
OPTIONS | Fetches all available REST operations | 200 | –
POST | Creates a new set of resources or a resource | 201 | 404, 409
PUT | Updates or replaces the given record | 200, 204 | 404
PATCH | Modifies the given record | 200, 204 | 404
DELETE | Deletes the given resource | 200 | 404

On the flip side, OpenSearch's existing APIs seem to use PUT to create the following resources:
* [Indices](https://github.com/opensearch-project/OpenSearch/blob/3f8e0cd3de219dee302788a3616ec33835055e5f/server/src/main/java/org/opensearch/rest/action/admin/indices/RestCreateIndexAction.java#L65)
* [Data streams](https://github.com/opensearch-project/OpenSearch/blob/15161a86491569d94642f3a88174e1389218b303/server/src/main/java/org/opensearch/rest/action/admin/indices/RestCreateDataStreamAction.java#L58)
* [Index templates](https://github.com/opensearch-project/OpenSearch/blob/3f8e0cd3de219dee302788a3616ec33835055e5f/server/src/main/java/org/opensearch/rest/action/admin/indices/RestPutIndexTemplateAction.java#L65)
* [Ingest pipelines](https://github.com/opensearch-project/OpenSearch/blob/aca2e9de7daaadee77c33e21ff5215c3f1e8600f/server/src/main/java/org/opensearch/rest/action/ingest/RestPutPipelineAction.java#L62)
* [Search pipelines](https://github.com/opensearch-project/OpenSearch/blob/aca2e9de7daaadee77c33e21ff5215c3f1e8600f/server/src/main/java/org/opensearch/rest/action/search/RestPutSearchPipelineAction.java#L38) (though, to be fair, I just copied whatever ingest pipelines did).

We use POST to create a [point-in-time context](https://github.com/opensearch-project/OpenSearch/blob/1e08b5a075fa6018be6c0af959b2d638c2743d56/server/src/main/java/org/opensearch/rest/action/search/RestCreatePitAction.java#L54).

We support *both* PUT and POST for creation of:
* [aliases](https://github.com/opensearch-project/OpenSearch/blob/1e08b5a075fa6018be6c0af959b2d638c2743d56/server/src/main/java/org/opensearch/rest/action/admin/indices/RestIndexPutAliasAction.java#L66), 
* [documents](https://github.com/opensearch-project/OpenSearch/blob/3f8e0cd3de219dee302788a3616ec33835055e5f/server/src/main/java/org/opensearch/rest/action/document/RestIndexAction.java#L65), 
* [snapshots](https://github.com/opensearch-project/OpenSearch/blob/60d7a098492c184e0b2a6fb0492d8c57b7626a54/server/src/main/java/org/opensearch/rest/action/admin/cluster/RestCreateSnapshotAction.java#L63), 
* [snapshot repositories](https://github.com/opensearch-project/OpenSearch/blob/9b9158e4cb692dfe04315225ce2b5f9c6c453307/server/src/main/java/org/opensearch/rest/action/admin/cluster/RestPutRepositoryAction.java#L63), and 
* [stored scripts](https://github.com/opensearch-project/OpenSearch/blob/aca2e9de7daaadee77c33e21ff5215c3f1e8600f/server/src/main/java/org/opensearch/rest/action/admin/cluster/RestPutStoredScriptAction.java#L65).

So, tl;dr, you are [technically correct](https://i.imgur.com/gRk1uZm.gif), but existing OpenSearch APIs are all over the place. I honestly don't know what would be least surprising for most users.